pkg/fetcher/definition_test.go

```diff
@@ -87,6 +91,133 @@ func TestFindMatching(t *testing.T) {
 
 	})
 
+	t.Run("return tests selected by label expressions", func(t *testing.T) {
+		// GIVEN
+		testA := &v1alpha1.TestDefinition{
+			ObjectMeta: v1.ObjectMeta{
+				UID:       uuid.NewUUID(),
+				Name:      "test-a",
+				Namespace: "test-a",
+				Labels: map[string]string{
+					"test": "true",
+				},
+			},
+		}
+		testB := &v1alpha1.TestDefinition{
+			ObjectMeta: v1.ObjectMeta{
+				UID:       uuid.NewUUID(),
+				Name:      "test-b",
+				Namespace: "test-b",
+				Labels: map[string]string{
+					"test": "false",
+				},
+			},
+		}
+		testC := &v1alpha1.TestDefinition{
+			ObjectMeta: v1.ObjectMeta{
+				UID:       uuid.NewUUID(),
+				Name:      "test-c",
+				Namespace: "test-c",
+				Labels: map[string]string{
+					"other": "123",
+				},
+			},
+		}
+
+		fakeCli := fake.NewFakeClientWithScheme(sch,
+			testA, testB, testC,
+		)
+		mockReader := &mockListReader{
+			fakeCli: fakeCli,
+			listResults: [][]v1alpha1.TestDefinition{
+				{*testC},
+				{*testA},
+			},
+		}
+		service := fetcher.NewForDefinition(mockReader)
+		// WHEN
+		out, err := service.FindMatching(v1alpha1.ClusterTestSuite{
+			Spec: v1alpha1.TestSuiteSpec{
+				Selectors: v1alpha1.TestsSelector{
+					MatchLabelExpressions: []string{
+						"other",
+						"test=true",
+					},
+				},
+			},
+		})
+		// THEN
+		require.NoError(t, err)
+		assert.Len(t, out, 2)
+		assert.Contains(t, out, *testA)
+		assert.Contains(t, out, *testC)
+	})
+
+	t.Run("return tests returns unique results", func(t *testing.T) {
```

Comment:
I don't understand this case. We should have for example test definition with 2 labels: 
`tier=backend`, `integration=true` and test suite that have 2 label expressions:
- tier=backend
- integration=true

And test should be listed only once

Reply:
The test checks if fetcher returns unique results across all selectors. I'll change it to reflect that more precisely.